### PR TITLE
132 using serializers

### LINKED ISSUE #132: DRF docs: Show APIView endpoints at the API root
- Register APIView urls with the DRF router

### PR DESCRIPTION
**Pull request by ecowan**

### CODE REVIEW

**Review by raprasad — DISMISSED** [review]
approved

File: server/opendp_apps/dataverses/serializers.py [hunk 1]
```diff
@@ -31,3 +31,9 @@ def save(self, **kwargs):
         self.validated_data['dv_general_token'] = dataverse_handoff.apiGeneralToken
         self.validated_data['dv_sensitive_token'] = dataverse_handoff.apiSensitiveDataReadToken
         return super().save()
+
```
Comment (raprasad) [inline server/opendp_apps/dataverses/serializers.py:39]: On line 25, for the DRF docs.
- switch: `fields = '__all__'` to 
   - `fields = ('dv_installation', 'user', 'dv_handoff')`

File: server/opendp_apps/dataverses/serializers.py [hunk 2]
```diff
@@ -31,3 +31,9 @@ def save(self, **kwargs):
...
+
+class DataverseHandoffSerializer(serializers.HyperlinkedModelSerializer):
+    class Meta:
+        model = DataverseHandoff
+        exclude = ['name', 'object_id', 'created']
```
Comment (raprasad) [inline server/opendp_apps/dataverses/serializers.py]: line 39: should `exclude` be `fields`?

File: server/opendp_apps/dataverses/views.py [hunk 3]
```diff
@@ -10,85 +10,91 @@
...
-from opendp_apps.dataverses.forms import DataverseHandoffForm
+from opendp_apps.dataverses.serializers import DataverseHandoffSerializer
 
 
-@login_required
-def view_dataverse_handoff(request):
-    """Temporarily save the Dataverse paramemeters +
-    redirect to the Vue page"""
-    if request.method == 'GET':
+class ViewDataverseHandoff(viewsets.ViewSet):
```
Comment (raprasad) [inline server/opendp_apps/dataverses/views.py]: should the viewsets be moved to separate files?

**Review by raprasad — CHANGES_REQUESTED** [review]
Included a couple of brief notes/questions

File: server/opendp_apps/dataverses/testing/test_views.py [hunk 4]
```diff
@@ -17,11 +17,13 @@ class TestDataverseUserView(BaseEndpointTest):
                 'test_opendp_users_01.json',
                 'test_dataverse_user_01.json']
 
-    def test_create(self):
+    def test_create(self, req_mocker):
+        self.set_mock_requests(req_mocker)
```
Comment (raprasad) [inline server/opendp_apps/dataverses/testing/test_views.py:15]: line 22:
- For the endpoint, there should only be two inputs: `user` and `dv_handoff`, e.g. remove dv_installation
```        
response = self.client.post('/api/dv-user/', data={'dv_installation': 1, 'user': 1, 'dv_handoff': 1})
```
- The inputs should query the models by `object_id`, not `id`

File: server/opendp_apps/dataverses/serializers.py [hunk 5]
```diff
@@ -34,7 +34,7 @@ def save(self, **kwargs):
         return super().save()
 
 
-class DataverseHandoffSerializer(serializers.HyperlinkedModelSerializer):
+class DataverseHandoffSerializer(serializers.ModelSerializer):
```
Comment (raprasad) [inline server/opendp_apps/dataverses/serializers.py:53]: Line 27: Needs only two fields. `fields = [ 'user', 'dv_handoff']`
 - (`dv_installation` is in the `dv_handoff` object)

**Review by raprasad — CHANGES_REQUESTED** [review]

File: server/opendp_apps/dataverses/serializers.py (hunk 5 above)
Comment (ecowan) [reply server/opendp_apps/dataverses/serializers.py:53]: `dv_installation` is not present in the `dv_handoff` object, though I am making this change now

**Review by raprasad — APPROVED** [review]
approved

**Pull request merged**